Engine: Ambiguity of time_t and int types on 32-bit archs

**Describe the bug**
After ea89d30, the ags engine no longer compiles on 32-bit architectures, because the `time_t` and `int` are equivalent 32-bit integer types. On 64-bit archs, the code compiles fine because `time_t` is a 64-bit integer while `int` remains a 32-bit integer.

The following contructors in Engine/ac/dynlib/scriptdatetime.h conflict:
`ScriptDateTime(const time_t &time);` and
`ScriptDateTime(int raw_time);`

```
c++ -std=c++11 -Werror=delete-non-virtual-dtor -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -isystem /usr/local/include  -fsigned-char -fno-strict-aliasing -fwrapv -Wunused-result -Wno-unused-value -Werror=write-strings -Werror=format -Werror=format-security -DNDEBUG -DALLEGRO_STATICLINK -DAL_LIBTYPE_STATIC -DLINUX_VERSION -DBUILTIN_PLUGINS -DHAVE_FSEEKO -DDISABLE_MPEG_AUDIO -DAGS_CASE_SENSITIVE_FILESYSTEM -D_FILE_OFFSET_BITS=64 -DRTLD_NEXT -I ../Common/libsrc/freetype-2.1.3/include -O2 -pipe  -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -I/usr/local/include -I/usr/local/include/SDL2 -D_REENTRANT -D_THREAD_SAFE -I/usr/local/include/SDL2 -I/usr/local/include -D_REENTRANT -D_THREAD_SAFE -I../Engine -I../Common -I../Common/libinclude -I../Common/libsrc/alfont-2.0.9 -I../Engine/libsrc/apeg-1.2.1/ -I../libsrc/mojoAL -I../Plugins -I../Engine/libsrc/glad/include -I../libsrc/miniz -I../libsrc/glm -I../libsrc/allegro/include -MD -c -o ac/datetime.o ac/datetime.cpp
ac/datetime.cpp:53:19: error: call to constructor of 'ScriptDateTime' is ambiguous
   53 |         sdt = new ScriptDateTime(raw_time);0
      |                   ^              ~~~~~~~~
../Engine/ac/dynobj/scriptdatetime.h:33:5: note: candidate constructor
   33 |     ScriptDateTime(const time_t &time);
      |     ^
../Engine/ac/dynobj/scriptdatetime.h:37:5: note: candidate constructor
   37 |     ScriptDateTime(int raw_time);
      |     ^
1 error generated.
```

**AGS Version**
3.6.2.13

**To Reproduce**
Build the ags engine on a system where `time_t` and `int` are equivalent integer types.

**Expected behavior**
Successful build.

**Desktop (please complete the following information):**
 - OS: FreeBSD
 - Version 14.2-i386 (64-bit varieties, e.g. amd64 build fine)

--- a/Engine/ac/datetime.cpp
+++ b/Engine/ac/datetime.cpp
@@ -35,7 +35,7 @@ ScriptDateTime* DateTime_CreateFromDate(int year, int month, int day, int hour, 
     }
     else
     {
-        sdt = new ScriptDateTime(year, month, day, hour, minute, second);
+        sdt = ScriptDateTime::FromFullDate(year, month, day, hour, minute, second);
     }
     ccRegisterManagedObject(sdt, sdt);
     return sdt;
@@ -50,7 +50,7 @@ ScriptDateTime* DateTime_CreateFromRawTime(int raw_time) {
     }
     else
     {
-        sdt = new ScriptDateTime(raw_time);
+        sdt = ScriptDateTime::FromRawTime(raw_time);
     }
     ccRegisterManagedObject(sdt, sdt);
     return sdt;

--- a/Engine/ac/dynobj/scriptdatetime.cpp
+++ b/Engine/ac/dynobj/scriptdatetime.cpp
@@ -31,22 +31,26 @@ const char *ScriptDateTime::GetType()
     return "DateTime";
 }
 
-ScriptDateTime::ScriptDateTime(const time_t &time)
-{
-    SetTime(SystemClock::from_time_t(time));
-}
-
 ScriptDateTime::ScriptDateTime(const ClockTimePoint &time)
 {
     SetTime(time);
 }
 
-ScriptDateTime::ScriptDateTime(int raw_time)
+/* static */ ScriptDateTime *ScriptDateTime::FromStdTime(const time_t &time)
 {
-    SetTime(ClockTimePoint(std::chrono::seconds(raw_time)));
+    ScriptDateTime *sdt = new ScriptDateTime();
+    sdt->SetTime(SystemClock::from_time_t(time));
+    return sdt;
 }
 
-ScriptDateTime::ScriptDateTime(int year, int month, int day, int hour, int minute, int second)
+/* static */ ScriptDateTime *ScriptDateTime::FromRawTime(int raw_time)
+{
+    ScriptDateTime *sdt = new ScriptDateTime();
+    sdt->SetTime(ClockTimePoint(std::chrono::seconds(raw_time)));
+    return sdt;
+}
+
+/* static */ ScriptDateTime *ScriptDateTime::FromFullDate(int year, int month, int day, int hour, int minute, int second)
 {
     // NOTE: we do not init our calendar fields here directly, and instead
     // go through SetTime, in case the combination of input values does not
@@ -60,7 +64,9 @@ ScriptDateTime::ScriptDateTime(int year, int month, int day, int hour, int minut
                    /* .tm_year = */ year - 1900,
                 };
     tm.tm_isdst = -1; // use DST value from local time zone
-    SetTime(std::chrono::system_clock::from_time_t(std::mktime(&tm)));
+    ScriptDateTime *sdt = new ScriptDateTime();
+    sdt->SetTime(std::chrono::system_clock::from_time_t(std::mktime(&tm)));
+    return sdt;
 }
 
 void ScriptDateTime::SetTime(const ClockTimePoint &time)

--- a/Engine/ac/dynobj/scriptdatetime.h
+++ b/Engine/ac/dynobj/scriptdatetime.h
@@ -29,14 +29,15 @@ public:
 
     // Constructs DateTime initialized with zero time
     ScriptDateTime() = default;
-    // Constructs DateTime initialized using C time_t
-    ScriptDateTime(const time_t &time);
     // Constructs DateTime initialized using chrono::time_point
     ScriptDateTime(const ClockTimePoint &time);
+
+    // Constructs DateTime initialized using C time_t
+    static ScriptDateTime *FromStdTime(const time_t &time);
     // Constructs DateTime initialized with raw time (unix time)
-    ScriptDateTime(int raw_time);
+    static ScriptDateTime *FromRawTime(int raw_time);
     // Constructs DateTime initialized with all the date/time components
-    ScriptDateTime(int year, int month, int day, int hour, int minute, int second);
+    static ScriptDateTime *FromFullDate(int year, int month, int day, int hour, int minute, int second);
 
     int Dispose(void *address, bool force) override;
     const char *GetType() override;

--- a/Engine/ac/file.cpp
+++ b/Engine/ac/file.cpp
@@ -71,7 +71,7 @@ ScriptDateTime* File_GetFileTime(const char *fnmm) {
     ft = File::GetFileTime(rp.FullPath);
   }
 
-  ScriptDateTime *sdt = new ScriptDateTime(ft);
+  ScriptDateTime *sdt = ScriptDateTime::FromStdTime(ft);
   ccRegisterManagedObject(sdt, sdt);
   return sdt;
 }

--- a/Engine/ac/game.cpp
+++ b/Engine/ac/game.cpp
@@ -423,7 +423,7 @@ const char* Game_GetSaveSlotDescription(int slnum) {
 ScriptDateTime* Game_GetSaveSlotTime(int slnum)
 {
     time_t ft = File::GetFileTime(get_save_game_path(slnum));
-    ScriptDateTime *sdt = new ScriptDateTime(ft);
+    ScriptDateTime *sdt = ScriptDateTime::FromStdTime(ft);
     ccRegisterManagedObject(sdt, sdt);
     return sdt;
 }

